rasa/shared/core/flows/flows_yaml_schema.json

```diff
@@ -0,0 +1,302 @@
+            }
+          },
+          {
+            "required": [
+              "collect"
+            ],
+            "additionalProperties": false,
+            "properties": {
+              "id": {
+                "type": "string"
+              },
+              "description":{
+                "type": "string"
+              },
+              "collect": {
+                "type": "string"
+              },
+              "ask_before_filling": {
+                "type": "boolean"
+              },
+              "reset_after_flow_ends": {
+                "type": "boolean"
+              },
+              "utter": {
+                "type": "string"
+              },
+              "rejections": {
+                "type": "array",
+                "items": {
+                  "type": "object",
+                  "required": [
+                    "if",
+                    "utter"
+                  ],
+                  "properties": {
+                    "if": {
+                      "type": "string"
+                    },
+                    "utter": {
+                      "type": "string"
+                    }
+                  }
+                }
+              },
+              "next": {
+                "$ref": "#$defs/next"
+              }
+            }
+          },
+          {
+            "required": [
+              "link"
+            ],
+            "additionalProperties": false,
+            "properties": {
+              "id": {
+                "type": "string"
+              },
+              "link": {
+                "type": "string"
+              },
+              "next": {
+                "type": "null"
+              }
+            }
+          },
+          {
+            "required": [
+              "set_slots"
+            ],
+            "additionalProperties": false,
+            "properties": {
+              "id": {
+                "type": "string"
+              },
+              "set_slots": {
+                "$ref": "#$defs/set_slots"
+              },
+              "next": {
+                "$ref": "#$defs/next"
+              }
+            }
+          },
+          {
+            "required": [
+              "next"
+            ],
+            "additionalProperties": false,
+            "properties": {
+              "next": {
+                "$ref": "#$defs/next"
+              },
+              "id": {
+                "type": "string"
+              }
+            }
+          },
+          {
+            "required": [
+              "generation_prompt"
+            ],
+            "additionalProperties": false,
+            "properties": {
+              "generation_prompt": {
+                "type": "string"
+              },
+              "id": {
+                "type": "string"
+              },
+              "next": {
+                "$ref": "#$defs/next"
+              }
+            }
+          },
+          {
+            "required": [
+              "intent"
```

Comment:
Just double-checking if this is still in use yet, or if it's been discarded, i thought we removed the need for steps of type user message, unless this is something different and related to coexistence?

Reply:
Added because of the flow spec specifically `3. Flow Rules` 3rd point it's not used anywhere (that I found) same with `nlu_trigger` maybe there are for same thing...

Reply:
Ok, yeah I was also under the impression that we don't use the `intent` step any longer, but `nlu_trigger` might be useful in coexistence. Best to check with Daksh/Engine on Slack if this part of the schema is correct.

Reply:
Yes we use `nlu_trigger` now. Although that property hasn't been implemented yet. @twerkmeister would know the implementation details of this.

Reply:
Thank for the clarification @dakshvar22 removed the `intent` step from the schema, the `nlu_trigger` is already part of the schema [here](https://github.com/RasaHQ/rasa/pull/12901/files#diff-6a18cabe4771845112a2b32f8e2518ddaec46e0068e084840d1e5edb637cfd35R191) (structure is the same as in the flows spec)